packages/aws-cdk-lib/aws-route53/lib/record-set.ts

```diff
@@ -930,7 +930,8 @@ export class CrossAccountZoneDelegationRecord extends Construct {
     }
 
     if (!props.delegatedZone.hostedZoneNameServers) {
-      throw Error(`Not able to retrieve Name Servers for ${props.delegatedZone.zoneName} due to it being imported.`);
```

Comment:
Instead of throwing here we could change this line linked from the original issue: https://github.com/aws/aws-cdk/blob/d69c51a2e10864e69d4610b6df30d40a9bfec3ac/packages/aws-cdk-lib/aws-route53/lib/cross-account-zone-delegation-handler/index.ts#L58

to read:

```
 ResourceRecords: (DelegatedZoneNameServers ?? []).map(ns => ({ Value: ns })), 
```

but that change would need to be carefully integ tested, we've recently seen issues from passing empty arrays around the custom resource handler framework.